addons/account/models/account_move.py

```diff
@@ -3180,23 +3180,23 @@ def add_file_data_results(file_data, invoice):
             if decoder:
                 try:
                     with self.env.cr.savepoint():
-                        with current_invoice._get_edi_creation() as invoice:
-                            # pylint: disable=not-callable
-                            success = decoder(invoice, file_data, new)
-                        if success or file_data['type'] == 'pdf':
-                            invoice._link_bill_origin_to_purchase_orders(timeout=4)
+                        # pylint: disable=not-callable
+                        success = decoder(current_invoice, file_data, new)
 
-                            invoices |= invoice
+                        if success or file_data['type'] == 'pdf':
+                            current_invoice = current_invoice or self.create({})
+                            current_invoice._link_bill_origin_to_purchase_orders(timeout=4)
+                            invoices |= current_invoice
+                            add_file_data_results(file_data, current_invoice)
                             current_invoice = self.env['account.move']
```

Comment:
In case of a decoder returning `False` before calling manually `_get_edi_creation`, `current_invoice` will still be an empty record. In that case, `current_invoice.sudo().message_post(body=message)` below would raise a traceback.
```suggestion
                        invoice = current_invoice or current_invoice.create({})
                        success = decoder(invoice, file_data, new)
                             
                        if success or file_data['type'] == 'pdf':
                            invoice._link_bill_origin_to_purchase_orders(timeout=4)
                            invoices |= invoice
                            current_invoice = self.env['account.move']
                            add_file_data_results(file_data, invoice)
```

Reply:
Okey thanks :) it's fixed now